### PR TITLE
add gluu-opa-pep plugin and test cases

### LINKED ISSUE #320: Make new OPA plugin
Open Policy Agent
https://www.openpolicyagent.org/

### PR DESCRIPTION
**Pull request by altexy**
Should close #320

### CODE REVIEW

File: kong/plugins/gluu-opa-pep/access.lua [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    local body = res.body
+    ngx.log(ngx.DEBUG, "OPA responds with body:\n", body)
+
+    local body_json, err = cjson2.decode(body)
+    if not body_json then
+        ngx.log(ngx.ERR, "JSON decode error: ", err)
+        kong.response.exit(502)
+    end
+
+    if body_json.result and body_json.result.allow then
```
Comment (ldeveloperl1985) [inline kong/plugins/gluu-opa-pep/access.lua]: One question: For this, we just need to add in docs `your policy must have to return the {result: {allow: true}} in successful authorization response`. right?

File: t/specs/gluu-opa-pep/test_spec.lua [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        },
+        modules = {
+            ["prometheus.lua"] = host_git_root .. "/third-party/nginx-lua-prometheus/prometheus.lua",
+            ["resty/lrucache.lua"] = host_git_root .. "/third-party/lua-resty-lrucache/lib/resty/lrucache.lua",
+            ["resty/lrucache/pureffi.lua"] = host_git_root .. "/third-party/lua-resty-lrucache/lib/resty/lrucache/pureffi.lua",
+        },
+        host_git_root = host_git_root,
+    }
+    kong_utils.backend()
+    kong_utils.opa()
```
Comment (ldeveloperl1985) [inline t/specs/gluu-opa-pep/test_spec.lua]: I think you forgot to commit  kong_utils file `opa()` function.

File: kong/plugins/gluu-opa-pep/access.lua (hunk 1 above)
Comment (altexy) [reply kong/plugins/gluu-opa-pep/access.lua]: it follows the pattern from https://www.openpolicyagent.org/docs/v0.10.7/http-api-authorization/
yes, we may mention it in the docs

File: kong/plugins/gluu-opa-pep/access.lua [hunk 3]
```diff
@@ -0,0 +1,75 @@
+local cjson = require "cjson.safe"
+local http = require "resty.http"
+
+local function path_plit(s)
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-opa-pep/access.lua]: done

File: kong/plugins/gluu-opa-pep/access.lua [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+end
+
+return function(self, conf)
+    local input = {}
+    -- current request info
+    input.method = ngx.req.get_method()
+    input.path = path_plit(ngx.var.uri) -- split normilized URI
+    input.uri_args = ngx.req.get_uri_args()
+    input.headers = ngx.req.get_headers()
+    -- input.body -- TODO do we need a body available for OPA? may be make it optional
```
[the remark this answers is not in the document]
Comment (altexy) [reply kong/plugins/gluu-opa-pep/access.lua]: done

**Comment by altexy** [comment]
@ldeveloperl1985 I have replaced the commit with all issues fixed

**Pull request merged**